generic/generic.go

```diff
@@ -0,0 +1,23 @@
+package generic
+
+// EqualsAny checks if a base value of type T is equal to
+// any of the other values of type T provided as arguments.
+func EqualsAny[T comparable](base T, all ...T) bool {
+	for _, v := range all {
+		if v == base {
+			return true
+		}
+	}
+	return false
+}
+
+// EqualsAll checks if a base value of type T is equal to all of the
+// other values of type T provided as arguments.
+func EqualsAll[T comparable](base T, all ...T) bool {
```

Comment:
we need to cover the case of `all` having zero items, in this case the function would return `true`, but the expected value is `false`